### ISSUE TITLE
Perpetual sync for block that doesn't exist yet

### ISSUE THREAD

**Issue by koen84** [ElrondNetwork/elrond-go#532]
Started a fresh observe, left it syncing, but it never arrives because it's syncing for a block that doesn't exist yet.

As can be seen in this first screenshot, it's trying to sync to 22757 block while 214?? is the highest final block sync.
![KV-Bug1a](https://user-images.githubusercontent.com/8461454/66635804-38e9f780-ec10-11e9-87c8-b3ec7215d18b.png)

After ctrl+c and restart ./node it immediately comes back as synchronised.  As can be seen in this second screenshot, from briefly after, it's now synced on 21496 block, significantly lower than which it was previously trying.
![KV-Bug1b](https://user-images.githubusercontent.com/8461454/66635996-93835380-ec10-11e9-90e7-eb79a20a599e.png)

It seems it was originally trying to sync to "cross check block height: meta", which is ahead.

Half an hour later the obervser is still reporting it's in sync, on 21746 block, still lower than what it was trying to sync to.  (See third screenshot.)
![KV-Bug1c](https://user-images.githubusercontent.com/8461454/66636383-62575300-ec11-11e9-9ce0-87fc7f852a42.png)

(Server time is UTC, as noted in the tmux right lower corner.)

**Comment by koen84** [comment]
Similar issue by another users on a windows node.  (Purple lines is what i want to point out.)

![KV-Bug1d](https://user-images.githubusercontent.com/8461454/66641731-98013980-ec1b-11e9-9698-1a3b78784323.jpg)

**Comment by iulianpascalau** [comment]
Please upload the full log of this observer.

**Comment by koen84** [comment]
@iulianpascalau looks like github complains about tgz and filesize, anywhere i could upload ?

**Comment by koen84** [comment]
Serving the tgz log archive over http here
[http://51.158.172.121:8888/KV-observerLogs-191011.tgz](url)

**Comment by koen84** [comment]
I observed this again with the security patch just now.  As my observer was affected and therefore got slightly out of sync and went into perpetual sync state with time only increasing.  I attached the 2 last logs here in github.

[0aaf3d2db238-2019-16-10-18-58-53.log](https://github.com/ElrondNetwork/elrond-go/files/3736013/0aaf3d2db238-2019-16-10-18-58-53.log)
[0aaf3d2db238-2019-16-10-19-16-36.log](https://github.com/ElrondNetwork/elrond-go/files/3736014/0aaf3d2db238-2019-16-10-19-16-36.log)
